### ISSUE TITLE
Python mapnik2 bindings broken

### ISSUE THREAD

**Issue by kaidamasaki** [Homebrew/homebrew-versions#1146]
After doing a `brew install mapnik2` opening a Python shell and importing mapnik results in the following.

```
$ python
Python 2.7.11 (default, Jan 22 2016, 08:29:18)
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import mapnik
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/mapnik/__init__.py", line 69, in <module>
    from _mapnik import *
ImportError: dlopen(/usr/local/lib/python2.7/site-packages/mapnik/_mapnik.so, 2): Symbol not found: __ZNK5boost9re_detail31cpp_regex_traits_implementationIcE17transform_primaryEPKcS4_
  Referenced from: /usr/local/Cellar/mapnik2/2.2.0_1/lib/libmapnik.dylib
  Expected in: /usr/local/opt/boost/lib/libboost_regex-mt.dylib
 in /usr/local/Cellar/mapnik2/2.2.0_1/lib/libmapnik.dylib
```

I've tried to debug this and swap out `boost155` but it looks like there's too many transitive dependencies on `boost` in Mapnik2 to make that simple.

**Comment by DomT4** [comment]
If you do `brew rm mapnik2 && brew install mapnik2 -s` does it reproduce?

**Comment by kaidamasaki** [comment]
`brew install mapnik2 -s` fails.

```
$ brew install mapnik2 -s
==> Installing mapnik2 from homebrew/versions
==> Downloading https://s3.amazonaws.com/mapnik/dist/v2.2.0/mapnik-v2.2.0.tar.bz
Already downloaded: /Library/Caches/Homebrew/mapnik2-2.2.0.tar.bz2
==> Downloading https://gist.githubusercontent.com/tdsmith/22aeb0bfb9691de91463/
Already downloaded: /Library/Caches/Homebrew/mapnik2--patch-40e83052ae892aa0b134c09d8610ebd891619895bb5f3e5d937d0c48ed42d1a6.diff
==> Patching
patching file bindings/python/mapnik_text_placement.cpp
patching file SConstruct
Hunk #1 succeeded at 1427 (offset -4 lines).
patching file include/mapnik/css_color_grammar.hpp
patching file include/mapnik/expression_grammar.hpp
patching file include/mapnik/expression_grammar_impl.hpp
patching file include/mapnik/json/feature_grammar.hpp
patching file include/mapnik/json/geometry_generator_grammar.hpp
patching file include/mapnik/json/geometry_grammar.hpp
patching file include/mapnik/svg/svg_path_commands.hpp
patching file include/mapnik/svg/svg_transform_grammar.hpp
patching file include/mapnik/util/geometry_svg_generator.hpp
patching file include/mapnik/util/geometry_wkt_generator.hpp
patching file include/mapnik/wkt/wkt_grammar.hpp
patching file src/json/feature_collection_parser.cpp
patching file plugins/input/geojson/geojson_datasource.cpp
patching file plugins/input/geojson/geojson_datasource.hpp
patching file plugins/input/geojson/geojson_featureset.cpp
patching file plugins/input/geojson/geojson_featureset.hpp
patching file src/save_map.cpp
patching file SConstruct
Hunk #1 succeeded at 1619 (offset -4 lines).
patching file include/mapnik/marker.hpp
==> python scons/scons.py configure CC="clang" CXX="clang++" JOBS=2 PREFIX=/usr/
==> python scons/scons.py install
Last 15 lines from /Users/kaida/Library/Logs/Homebrew/mapnik2/02.python:
clang: warning: optimization flag '-finline-functions' is not supported
clang: warning: argument unused during compilation: '-finline-functions'
In file included from src/expression_node.cpp:23:
In file included from include/mapnik/expression_node.hpp:27:
In file included from include/mapnik/value.hpp:33:
In file included from /usr/local/Cellar/boost/1.60.0_1/include/boost/variant/recursive_wrapper.hpp:16:
/usr/local/Cellar/boost/1.60.0_1/include/boost/variant/recursive_wrapper_fwd.hpp:53:12: error: expected class name
    : mpl::false_
           ^
/usr/local/Cellar/boost/1.60.0_1/include/boost/variant/recursive_wrapper_fwd.hpp:59:12: error: expected class name
    : mpl::true_
           ^
2 errors generated.
scons: *** [src/expression_node.os] Error 1
scons: building terminated because of errors.

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/homebrew):
  https://github.com/Homebrew/homebrew-versions/issues
```

**Comment by gordonk** [comment]
Ever get this fixed?

**Comment by mikemcquaid** [comment]
@gordonk If it did the issue would be closed.

**Comment by gordonk** [comment]
@mikemcquaid My apologies.  Let me rephrase.

@kaidamasaki Ever find a work around?

**Comment by kaidamasaki** [comment]
My workaround is to use an Ubuntu VM...

**Comment by gordonk** [comment]
Heh.  Luckily I've got all that working, just coming back to an old project and couldn't get it to compile to do some development.

**Comment by DomT4** [comment]
Fixed. Gave up on upstream continually backporting more and more Boost fixes and just pegged it to the 1.59.0 release, which is the last one it didn't die horribly against.

**Comment by kaidamasaki** [comment]
Thanks!
